### DISCUSSION TITLE
Issue accessing GitInfo variables for Multi-targeted project in MSBuild when packing into nuget

### DISCUSSION THREAD

**Discussion by kikaragyozov** [devlooped/GitInfo#221] [Q&A]
The project targets the following frameworks:
```
<TargetFrameworks>netstandard2.0;net6.0</TargetFrameworks>
```

I have the following target defined in my .csproj file:

```
<Target Name="PopulateInfo" DependsOnTargets="GitInfo" BeforeTargets="PrepareForBuild">
	<PropertyGroup>
		<Version>$(GitBaseVersionMajor).$(GitBaseVersionMinor).$(GitBaseVersionPatch).0</Version>
		<FileVersion>$(GitBaseVersionMajor).$(GitBaseVersionMinor).$(GitBaseVersionPatch).$(GitCommits)</FileVersion>
		<PackageVersion>$(FileVersion)</PackageVersion>
		<InformationalVersion>$(FileVersion)+branch.$(GitBranch).commit.$(GitCommit)</InformationalVersion>
	</PropertyGroup>
</Target>
```
I'm trying to build & pack a nuget package, but it always ends up with the same version of 1.0.0.

I'm using the following command to try and build my package: 

```
MSBuild.exe /t:Clean;Build;Pack /p:Configuration=Release
```

**Comment by kzu** [comment]
Multitargeting build does not have a `PrepareForBuild` target, so your target will never run.

See https://github.com/dotnet/msbuild/blob/main/src/Tasks/Microsoft.Common.CrossTargeting.targets and the NuGet targets at https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets (which are imported for both single and multi-targeting builds).

You can update the target as follows to run in both scenarios:

```xml
<Target Name="PopulateNETSDKAssemblyVersion" BeforeTargets="CoreCompile;GenerateNuspec;GetPackageContents" DependsOnTargets="GitVersion">
```

> NOTE: GetPackageContents makes it compatible with [NuGetizer](https://nuget.org/packages/NuGetizer) which I'd highly recommend for packaging, but I'm obviously biased ;))
